addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -580,17 +580,19 @@ export class X2ManyFieldDialog extends Component {
             this.archInfo.arch = this.archInfo.xmlDoc.outerHTML;
         }
 
-        const { autofocusFieldId, disableAutofocus } = this.archInfo;
+        const { autofocusFieldIds, disableAutofocus } = this.archInfo;
         if (!disableAutofocus) {
             // to simplify
             useEffect(
                 (isInEdition) => {
                     let elementToFocus;
                     if (isInEdition) {
-                        elementToFocus =
-                            (autofocusFieldId &&
-                                this.modalRef.el.querySelector(`#${autofocusFieldId}`)) ||
-                            this.modalRef.el.querySelector(".o_field_widget input");
+                        const focusableSelectors = ".o_field_widget input";
+                        elementToFocus = autofocusFieldIds?.map(id => this.modalRef.el.querySelector(`#${id}`))
+                            .find(el => el)
+                            || this.modalRef.el.querySelector(
+                                focusableSelectors
+                            );
```

Comment:
same as below (and I noticed a that the `focusableSelectors` variable is not really useful here